### PR TITLE
DAOS-7174 doc: refine reintegration section

### PR DESCRIPTION
**Pull request by johannlombardi**
Provide more details on reintegration and document the current
limitation.
Add a more detailed definition of what a container is and
how it can be used.
Fix a few nits where UUID are still used on the cmd line.

doc-only: true

Signed-off-by: Johann Lombardi <johann.lombardi@intel.com>

### CODE REVIEW

**Comment by johannlombardi** [comment]
Please review.

File: docs/admin/administration.md [hunk 1]
```diff
@@ -507,22 +502,38 @@ UUID, in addition to the rank state.
...
+operate and listen on the management network.
+
+!!! warning
+    All engines monitor each other and pro-actively exclude unresponsive
+    members. It is critical to properly stop a DAOS system as with dmg in
+    the case of a planned maintenance on all or a majority of the DAOS
+    storage nodes. An abrupt reboot of the storage nodes might result
+    in massive exclusion that will take time to recover.
+
+The force option can be passed to dmg system stop for cases when a clean
```
Comment (cdurf1) [inline docs/admin/administration.md:519]: maybe change to:
The force option can be passed to the dmg system stop when a clean shutown is not working. Monitoring is not disabled in this case, and spurious exclusion might happen, but the engines are guaranteed to be killed.

File: docs/admin/administration.md [hunk 2]
```diff
@@ -532,9 +543,21 @@ The output table will indicate action and result.
 
 DAOS I/O Engines will be started.
 
+As for shutdow, a list of engines to restart can be specified on the command
```
Comment (cdurf1) [inline docs/admin/administration.md:546]: shutdown

File: docs/admin/administration.md [hunk 3]
```diff
@@ -532,9 +543,21 @@ The output table will indicate action and result.
...
+line:
+
+`$ dmg system start [--ranks <rankset>|--host-ranks <hostset>]`
+
+- `<rankset>` is a pattern describing rank ranges e.g. 0,5-10,20-100
```
Comment (cdurf1) [inline docs/admin/administration.md:551]: e.g.,

File: docs/admin/administration.md [hunk 4]
```diff
@@ -532,9 +543,21 @@ The output table will indicate action and result.
...
+- `<hostset>` is a pattern describing host ranges e.g.
```
Comment (cdurf1) [inline docs/admin/administration.md:552]: e.g.,

File: docs/admin/administration.md [hunk 5]
```diff
@@ -532,9 +543,21 @@ The output table will indicate action and result.
...
+storagehost[0,5-10],10.8.1.[20-100]
+
+If the ranks were excluded from pools (e.g. unclean shutdown), they will need to
```
Comment (cdurf1) [inline docs/admin/administration.md:555]: e.g.,

File: docs/admin/pool_operations.md [hunk 6]
```diff
@@ -417,12 +417,12 @@ will be decided based on the remaining ACL rules.
 
 ### Exclusion & Self-healing
 
-An operator can exclude one or more targets from a specific DAOS pool using the rank
-the target resides on as well as the target idx on that rank. If a target idx list is
-not provided then all targets on the rank will be excluded. Excluding a target will
-automatically start the rebuild process.
+An operator can exclude one or more engines or targets from a specific DAOS pool
+using the rank the target resides on as well as the target idx on that rank.
```
Comment (cdurf1) [inline docs/admin/pool_operations.md:421]: ...using the rank that the target resides, as well as the target idx on that rank.

File: docs/admin/pool_operations.md [hunk 7]
```diff
@@ -417,12 +417,12 @@ will be decided based on the remaining ACL rules.
...
+If a target idx list is not provided then all targets on the rank will be excluded.
```
Comment (cdurf1) [inline docs/admin/pool_operations.md:422]: ...not provided, all targets on the rank ...

File: docs/admin/pool_operations.md [hunk 8]
```diff
@@ -435,15 +435,17 @@ The pool target exclude command accepts 2 parameters:
 
 ### Drain
 
-Alternatively when an operator would like to remove one or more pool targets
-without the system operating in degraded mode Drain can be used. A pool drain operation will
-initiate rebuild without excluding the designated target until after the rebuild is complete.
-This allows the target(s) drained to continue to perform I/O while the rebuild
+Alternatively when an operator would like to remove one or more engines or
```
Comment (cdurf1) [inline docs/admin/pool_operations.md:438]: Alternatively,

File: docs/admin/pool_operations.md [hunk 9]
```diff
@@ -456,20 +458,21 @@ The pool target drain command accepts 2 parameters:
 
 ### Reintegration
 
-After a target failure an operator can fix the underlying issue and reintegrate the
-affected targets to restore the pool to its original state. The operator can either
-reintegrate specific targets for a rank by supplying a target idx list, or reintegrate
-an entire rank by omitting the list.
+After a engine failure and exclusion, an operator can fix the underlying issue
```
Comment (cdurf1) [inline docs/admin/pool_operations.md:461]: an engine

File: docs/admin/pool_operations.md [hunk 10]
```diff
@@ -488,6 +491,14 @@ These should be the same values used when reintegrating the targets.
 $ dmg pool reintegrate --rank=5 --target-idx=0,1 <pool_label>
 ```
 
+!!! warning
+    While dmg pool query and list show how many targets are disabled for each
+    pool, there is currently no way to list the targets that have actually
+    been disabled. As a result, it is recommended for now to try to reintegrate
+    all engine ranks one after the other via `for i in seq $NR_RANKs; do dmg
+    pool reintegrate --rank=$i; done`. This limitation will be address in the
```
Comment (cdurf1) [inline docs/admin/pool_operations.md:499]: ...will be addressed

File: docs/user/container.md [hunk 11]
```diff
@@ -1,21 +1,31 @@
...
-POSIX namespace, an HDF5 file or any other application-specific data model.
-The chapter explains how to manage a container, while the subsequent ones detail
-how to access a DAOS container from applications.
+DAOS containers are datasets managed by the users. Similarly to S3 buckets,
+a DAOS container is a collection of objects that can be presented to applications
+through different interfaces including POSIX I/O (files and directories),
+HDF5, SQL or any other data models of your choice.
 
-## Container Basic Operations
+A container belongs to a single pool and share the space with other containers.
```
Comment (cdurf1) [inline docs/user/container.md:8]: ...and shares

File: docs/user/container.md [hunk 12]
```diff
@@ -1,21 +1,31 @@
...
+It is identified by a label selected by the user and a immutable UUID allocated
```
Comment (cdurf1) [inline docs/user/container.md:9]: ...an immutable

File: docs/user/container.md [hunk 13]
```diff
@@ -1,21 +1,31 @@
...
+when the container is first created.
+
+The container is the unit of data management in DAOS and can be snapshotted or
+cloned.
+
+!!! warning
+    DAOS containers are storage containers should not be confused with Linux
```
Comment (cdurf1) [inline docs/user/container.md:16]: ...and should

File: docs/user/container.md [hunk 14]
```diff
@@ -1,21 +1,31 @@
...
+    containers.
+
+## Container Basics
+
+Containers can be created, queried, relabled, listed and destroyed through the
```
Comment (cdurf1) [inline docs/user/container.md:21]: , relabeled,

File: docs/user/container.md [hunk 15]
```diff
@@ -88,10 +100,17 @@ $ daos cont destroy --path /tmp/mycontainer
 Successfully destroyed container 30e5d364-62c9-4ddf-9284-1021359455f2
 ```
 
+If the container is in use, the force option (i.e. --force or -f) must be added.
```
Comment (cdurf1) [inline docs/user/container.md:103]: i.e,

File: docs/user/container.md [hunk 16]
```diff
@@ -88,10 +100,17 @@ $ daos cont destroy --path /tmp/mycontainer
...
+Active users of a force-deleted container will fail with bad handle error.
```
Comment (cdurf1) [inline docs/user/container.md:104]: ...fail with a bad handle error

**Comment by johannlombardi** [comment]
Have addressed all comments from Craig. Please review.

**Pull request merged**